docs/object_stores.md

```diff
@@ -0,0 +1,23 @@
+# Object Storages
+
+Thanos supports any object stores that can be implemented against Thanos [objstore.Bucket inteface](/pkg/objstore/objstore.go)
+
+Current object storage client implementations:
+
+| Provider             | Maturity | Auto-tested on CI | Maintainers |
+|----------------------|-------------------|-----------|---------------|
+| Google Cloud Storage | Stable  (production usage)             | yes       | @bplotka   |
+| AWS S3               | Beta  (working PoCs, testing usage)               | no        | ?          |
+
+NOTE: Currently Thanos requires strong consistency (write-read) for object store implementation.
+
+## How to add a new client?
+
+1. Create new directory under `pkg/objstore/<provider>`
+2. Implement [objstore.Bucket inteface](/pkg/objstore/objstore.go)
+3. Add `NewTestBucket` constructor for testing purposes, that creates and deletes temporary bucket.
+4. Use created `NewTestBucket` in [ForeachStore method](/pkg/objstore/objtesting/foreach.go) to ensure we can run tests (optionally) against new provider.
```

Comment:
Why `(optionally)` ? Why don't you mandate the testing of the new provider?

Reply:
Yea, this is mandatory, just I meant that to run this test (see https://github.com/improbable-eng/thanos/pull/327) you need to have literally manually set envvars to your provider env. Otherwise this test will be skipped (no one wants to have deployed all the providers to run single test) Maybe we can work on wording here.